### PR TITLE
Setup some gitStream actions against JVM team PRs to assess this tool

### PR DESCRIPTION
**Pull request by tresat**
The actions I've added will:

- Categorize change with labels for each platform (for only a subset of directories, not the complete list yet) and with a label if the build script is touched. 
- Add table describing code experts in a comment.
- Make complex changes (>20 files, > 45 estimated time to review) require 2 approvals.
- Add estimated time to review label.
- If a PR includes TODOs, add a comment announcing this and request changes.
- If a PR is a source change that lacks tests, add a comment announcing this.

These are incomplete but can be expanded and modified later.  This PR is aimed at getting a variety of actions up and running so we can see if they are helpful.

### CODE REVIEW

File: .cm/change_categories.cm [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# Read the "|" not as "or", but as a "pipe", taking the output of the previous command and passing it to the next command.
+
+# The results of these calculations are assigned to `is.formatting`, `is.docs` and `is.tests`.
+is:
+  formatting: {{ source.diff.files | isFormattingChange }}
+  docs: {{ files | allDocs }}
+  tests: {{ files | allTests }}
+
+  # This is not a complete list, just a demonstration
+  # Perhaps a task could generate this list from the CODEOWNERS, or both this and CODEOWNERS could be generated from different single source of truth?
```
Comment (big-guy) [inline .cm/change_categories.cm:90]: Does gitStream have some native way to get the owners from CODEOWNERS or could we have our own way of using CODEOWNERS as the source of truth?

**Review by big-guy — APPROVED** [review]

File: .cm/lacks_tests.cm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if:
+      - {{ author.on_jvm_team }}
+      - {{ includes_src_changes }}
+      - {{ not (includes_test_changes) }}
+      - {{ not (is_formatting_only_change) }} # Not only formatting-only changes
+    run:
+      - action: add-comment@v1
+        args:
+          comment: |
+            'This PR appears to be lacking tests.  Consider adding tests to cover the new functionality.'
```
Comment (big-guy) [inline .cm/lacks_tests.cm:31]: It looks like this includes the quotes in the message? https://github.com/tresat/test-git-stream/pull/25#issuecomment-1505882863

File: .cm/includes_todos.cm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  # If someone adds a TODO, remind them about it by requesting removal
+  point_out_todos:
+    if:
+      - {{ author.on_jvm_team }}
+      - {{ source.diff.files | matchDiffLines(regex=r/(TODO)|(todo)/) | some }}
+    run:
+      - action: request-changes@v1
+        args:
+          comment: 'There are new TODOs present in this change.  Should any be removed?'
```
[the remark this answers is not in the document]
Comment (tresat) [reply .cm/includes_todos.cm:28]: There's no way to do this yet.  I've submitted a [feature request](https://github.com/linear-b/gitstream/issues/90).

File: .cm/change_categories.cm (hunk 1 above)
Comment (tresat) [reply .cm/change_categories.cm:90]: Not yet.  Custom filter functions are coming soon.  This [unpublished docs page in gitstream's github repo ](https://github.com/linear-b/gitstream/blob/main/docs/custom-filters.md) describes how you'll be able to write them in javascript, and even use npm packages.  

Once this is possible, sourcing the data from `CODEOWNERS` should be straightforward.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**